pygaggle/run/evaluate_trec_covid_ranker.py

```diff
@@ -0,0 +1,175 @@
+                  'random')
+
+class DocumentRankingEvaluationOptions(BaseModel):
+    task: str
+    dataset: Path
+    index_dir: Path
+    method: str
+    model: str
+    split: str
+    batch_size: int
+    seg_size: int
+    seg_stride: int
+    aggregate_method: str
+    device: str
+    is_duo: bool
+    from_tf: bool
+    metrics: List[str]
+    model_type: Optional[str]
+    tokenizer_name: Optional[str]
+
+    @validator('task')
+    def task_exists(cls, v: str):
+        assert v in ['trec-covid']
+
+    @validator('dataset')
+    def dataset_exists(cls, v: Path):
+        assert v.exists(), 'data directory must exist'
+        return v
+
+    @validator('index_dir')
+    def index_dir_exists(cls, v: Path):
+        assert v.exists(), 'index directory must exist'
+        return v
+
+    @validator('model')
+    def model_sane(cls, v: str, values, **kwargs):
+        method = values['method']
+        if method == 'transformer' and v is None:
+            raise ValueError('transformer name or path must be specified')
+        return v
+
+    @validator('tokenizer_name')
+    def tokenizer_sane(cls, v: str, values, **kwargs):
+        if v is None:
+            return values['model']
+        return v
+
+
+def construct_t5(options: DocumentRankingEvaluationOptions) -> Reranker:
+    device = torch.device(options.device)
+    model = T5ForConditionalGeneration.from_pretrained(options.model,
+                                                       from_tf=options.from_tf).to(device).eval()
+    tokenizer = AutoTokenizer.from_pretrained(options.model_type)
+    tokenizer = T5BatchTokenizer(tokenizer, options.batch_size)
+    return T5Reranker(model, tokenizer)
+
+
+def construct_transformer(options:
+                          DocumentRankingEvaluationOptions) -> Reranker:
+    device = torch.device(options.device)
+    model = AutoModel.from_pretrained(options.model,
+                                      from_tf=options.from_tf).to(device).eval()
+    tokenizer = SimpleBatchTokenizer(AutoTokenizer.from_pretrained(
+        options.tokenizer_name),
+        options.batch_size)
+    provider = CosineSimilarityMatrixProvider()
+    return UnsupervisedTransformerReranker(model, tokenizer, provider)
+
+
+def construct_seq_class_transformer(options: DocumentRankingEvaluationOptions
+                                    ) -> Reranker:
+    model = AutoModelForSequenceClassification.from_pretrained(options.model, from_tf=options.from_tf)
+    device = torch.device(options.device)
+    model = model.to(device).eval()
+    tokenizer = AutoTokenizer.from_pretrained(options.tokenizer_name)
+    return SequenceClassificationTransformerReranker(model, tokenizer)
+
+
+def construct_bm25(options: DocumentRankingEvaluationOptions) -> Reranker:
+    return Bm25Reranker(index_path=str(options.index_dir))
+
+
+def main():
+    apb = ArgumentParserBuilder()
+    apb.add_opts(opt('--task',
```

Comment:
Add descriptions for the parameters?

Reply:
Just to be clear, do you mean setting the `help` string for each or just describing the parameters as comments in the code?

Reply:
for each parameter